src/fpm.f90

```diff
@@ -62,6 +62,23 @@ subroutine build_model(model, settings, package, error)
         model%fortran_compiler = settings%compiler
     endif
 
+    associate(os_type => get_os_type())
+        if (os_type /= OS_WINDOWS .and. os_type /= OS_UNKNOWN) then
+            model%archiver = "ar -rs "
+        else
+            block
+                integer :: estat
+
+                call execute_command_line("ar --version", exitstat=estat)
+                if (estat /= 0) then
+                    model%archiver = "lib /OUT:"
+                else
+                    model%archiver = "ar -rs "
+                end if
+            end block
+        end if
+    end associate
```

Comment:
Why not make it a function?

Reply:
Good point. I'll extract it into a function, `get_archiver`.